src/org/parosproxy/paros/extension/state/ExtensionState.java

```diff
@@ -67,11 +75,46 @@ public void hook(ExtensionHook extensionHook) {
 	    // ZAP: Changed to check if there is a view.
 	    if (getView() != null) {
 		    extensionHook.getHookMenu().addEditMenuItem(extensionHook.getHookMenu().getMenuSeparator());
+		    extensionHook.getHookMenu().addEditMenuItem(getMenuZAPMode());
 		    extensionHook.getHookMenu().addEditMenuItem(getMenuSessionTrackingEnable());
 		    extensionHook.getHookMenu().addEditMenuItem(getMenuResetSessionState());
 	    }
 	}
 
+	private JMenuItem getMenuZAPMode(){
+		if (menuZAPMode == null) {
+			menuZAPMode = new JMenu(Constant.messages.getString("menu.edit.zapmode"));
+			int modeCount = 0;
+			for (Mode modeType : Mode.values()) {
+			menuItemsMode[modeCount] = addZAPModeMenuItem(modeType);
+  		menuZAPMode.add(menuItemsMode[modeCount]);
+			modeCount++;
+			}
+			modeCount = 0;
+			for(Mode modeType : Mode.values()){
+			addZAPModeEventListeners(modeType,menuItemsMode[modeCount],menuItemsMode);
+			modeCount++;
+			}
+		}
+		return menuZAPMode;
+	}
+
+	private void addZAPModeEventListeners(final Mode modeType, final JCheckBoxMenuItem checkBox, final JCheckBoxMenuItem[] menuItemsMode){
+		 checkBox.addActionListener(new java.awt.event.ActionListener() {
+		 public void actionPerformed(java.awt.event.ActionEvent evt) {

```

Comment:
I don't have an IDE in front of me.  But, aren't you overriding actionPerformed on the checkbox? I'm kind of trusting google here (since no IDE).... http://www.codejava.net/java-se/swing/jcheckbox-basic-tutorial-and-examples#AddListeners